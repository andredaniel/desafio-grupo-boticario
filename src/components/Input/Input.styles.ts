import styled from 'styled-components'
import { theme } from '../../theme'

export const InputWrapper = styled.div`
  padding: 0.5rem 0;
  width: 100%;

  label {
    display: block;
    margin-bottom: 5px;
  }
  
  input {
    background: #fff;
    border-radius: 0.4rem;
    border: 1px solid #ddd;
    padding: 0.7rem 1rem;
    width: 100%;
    font-size: 1rem;

    &:focus {
      outline: none;
      border-color: ${theme.colors.primary.MAIN};
    }
  }
`
