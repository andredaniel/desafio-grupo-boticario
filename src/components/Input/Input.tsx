import React, { InputHTMLAttributes } from 'react'
import InputMask from 'react-input-mask'
import { InputWrapper } from './Input.styles'

interface IInput extends InputHTMLAttributes<HTMLInputElement> {
  label?: string
  mask?: string
}

const Input: React.FC<IInput> = (props): JSX.Element => {
  const { label, id, type, mask = '', ...rest } = props

  return (
    <InputWrapper>
      {label && <label htmlFor={id}>{label}</label>}
      <InputMask
        id={id}
        mask={mask}
        maskPlaceholder={null}
        type={type || 'text'}
        {...rest}
      />
    </InputWrapper>
  )
}

export { Input }
export type { IInput }
