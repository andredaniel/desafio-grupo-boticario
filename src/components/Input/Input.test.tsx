import { render } from '@testing-library/react'
import React from 'react'
import TestUtils from 'react-dom/test-utils'
import { Input } from './Input'

const changeInputMaskValue = (element: any, value: any) => {
  element.value = value
  element.selectionStart = element.selectionEnd = value.length
  TestUtils.Simulate.change(element)
}

describe('<Input />', () => {
  it('should render an Input without mask', () => {
    const { container } = render(<Input />)
    expect(container).toMatchSnapshot()
  })

  it('should render an Input with mask', () => {
    const component = <Input mask="999.999.999-99" placeholder="cpf" />
    const { getByPlaceholderText } = render(component)
    const input = getByPlaceholderText('cpf')

    changeInputMaskValue(input, '12345678945')

    expect(input).toHaveValue('123.456.789-45')
  })
})

export { changeInputMaskValue }
