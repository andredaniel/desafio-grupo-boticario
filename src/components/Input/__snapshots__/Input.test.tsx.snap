// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render an Input without mask 1`] = `
<div>
  <div
    class="sc-bdfBwQ jSMxIb"
  >
    <input
      type="text"
    />
  </div>
</div>
`;
